pkg/controller/infrastructure/configvalidator.go

```diff
@@ -76,7 +76,12 @@ func (c *configValidator) Validate(ctx context.Context, infra *extensionsv1alpha
 func (c *configValidator) validateNetworks(ctx context.Context, computeClient gcpclient.ComputeClient, clusterName, region string, networks api.NetworkConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if networks.CloudNAT == nil || len(networks.CloudNAT.NatIPNames) == 0 {
+	if networks.CloudNAT == nil {
+		return allErrs
+	}
+
+	if len(networks.CloudNAT.NatIPNames) == 0 {
```

Comment:
The validation should be rather part of `admission-gcp` - https://github.com/gardener/gardener-extension-provider-gcp/blob/37e1e2ca92132db555271ed286f8d3d337253085/pkg/apis/gcp/validation/infrastructure.go#L28 . 

This is to prevent created/updating Shoots with empty `networks.cloudNAT.natIPNames`.
The `ConfigValidator` validation is executed before Infrastructure reconciliation when the Shoot is already created/updated with the invalid spec. And the `ConfigValidator` was mainly introduced to allow validating resources using the cloud provider APIs as well. You can read more about the concept in https://github.com/gardener/gardener/blob/v1.58.0/docs/extensions/infrastructure.md#configvalidator-interface.